### DISCUSSION TITLE
v0.8.0

### DISCUSSION THREAD

**Discussion by linonetwo** [tiddly-gittly/TidGi-Desktop#424] [Release]
## Notice

For users using HTTP API, you need to turn on the http server feature in Workspace Settings - Server Settings, which is now turned off by default to improve startup speed.

Many of the new features added this time, such as Taiji's direct blog opening, are in the right-click workspace icon - Workspace Settings - Server Settings.

## What's Changed

- chore: add appimage build @SpacingBat3
- chore: upgrade deps
- chore: Cancel the win32 build for now until someone needs it
- Close some issues, and increase the performance.
- docs: better button text about save&restart in workspace settings
- Feat: add Sqlite Database, not used, for developer to test
- feat: allow file:// protocol in wiki link
- feat: allow lazy-all on workspace config
- Feat: allow zx script pass variable between nodejs and tiddlywiki context by @linonetwo in https://github.com/tiddly-gittly/TidGi-Desktop/pull/407
- feat: https mode for blog server
- feat: let workspace setting save button clearer
- feat: menu item to ExportWholeWikiHTML
- feat: readonly mode
- feat: set excludedPlugins on wiki startup, on blog settings in workspace settings
- feat: show lastNodeJSArgv in workspace settings
- feat: support file relative to wiki folder
- feat: sync all subwiki together, if main workspace set to sync
- feat: TokenAuth: token based access protection in LAN
- feat: use ipc load html, prevent http error, and load faster by @linonetwo in https://github.com/tiddly-gittly/TidGi-Desktop/pull/417
- feat: use wiki workspace name as saved html name
- feat: use workspace id for wiki worker, instead of path, so you can start multiple workspace on a folder
- fix #369 降级Ubuntu版本适配Deepin包管理工具 by @Zacharia2 in https://github.com/tiddly-gittly/TidGi-Desktop/pull/372
- fix: Cannot read properties of null (reading 'send')
- fix: ctrl + r to refresh wiki
- fix: don't dereference dugite, to reduce macos build size
- fix: empty json config file crush app
- fix: error about no adminToken
- fix: Error when import/export HTML wiki: The "path" argument must be of type string. Received undefined (b…
- fix: git worker uncatched error crush app, and untranslated error message
- fix: hibernated workspace still have active view
- fix: input-box lose focus after wiki shows dialog (that calls window.alert('...') )
- fix: only set language on create new wiki (not when clone)
- fix: prevent in some rare case webContents is null
- fix: save and restore menubar window position
- fix: sendToMainWindowAndAwait error when app on background
- fix: some config didn't trigger restart
- fix: view still active after remove wiki
- fix: zoom menu item (ctrl + +/- to zoom)
- Fix: zx script not working
- refactor: add more operations to wikiOperations from `src/preload/wikiOperation.ts`
- Refactor: can use file protocol for link to open folder and load image by @linonetwo in https://github.com/tiddly-gittly/TidGi-Desktop/pull/421
- refactor: make adminToken usage clearer to fix potential bugs
- refactor: move blog-related options to a accordion in workspace settings
- refactor: use workspaceID instead of wikiFolderLocation in wiki methods (this may cause bug, need more test)

## 注意

使用 HTTP API 的用户，需要在工作区设置-服务器设置里打开http服务器功能，现在默认关了，以提高启动速度。

本次新增的很多功能，比如太记直接开博客，都在右键工作区图标-工作区设置-服务器设置里。

## 本次更新

- 杂项：add appimage build @SpacingBat3
- 杂项：升级软件依赖
- 杂项：暂时取消 win32 构建直到有人需要它
- 关闭一些 issue，并提高性能。
- 文档：在工作区设置中加入更好的关于保存和重启的按钮文本
- 功能：添加Sqlite数据库，不使用，供开发人员测试。
- 功能：允许在维基链接中使用 file:// 协议
- 功能：允许在工作区配置中使用lazy-all功能
- 功能：允许zx脚本在nodejs和tiddlywiki之间传递变量 由 @linonetwo 在https://github.com/tiddly-gittly/TidGi-Desktop/pull/407
- 功能：博客服务器的https模式
- 功能：让工作区设置的保存按钮更清晰
- 功能：导出整个WikiHTML的菜单项
- 功能：只读模式
- 功能：在wiki启动时设置排除插件，在工作区设置中设置博客设置
- 功能：在工作区设置中显示lastNodeJSArgv
- 功能：支持相对于维基文件夹的文件
- 功能：如果主工作区设置为同步，则同步所有子维基
- 功能：凭证鉴权：局域网中基于凭证的访问保护
- 功能：使用ipc加载html，防止http错误，并在 https://github.com/tiddly-gittly/TidGi-Desktop/pull/417 ，由 @linonetwo 加载更快。
- 功能：使用维基工作区名称作为保存的HTML名称
- 功能：为维基工作者使用工作空间ID，而不是路径，这样你就可以在一个文件夹上启动多个工作空间
- 修复 #369 降级Ubuntu版本适配Deepin包管理工具 由 @Zacharia2 在 https://github.com/tiddly-gittly/TidGi-Desktop/pull/372 发表
- 修复： 无法读取null的属性（读取'发送'）。
- 修复：用ctrl+r刷新维基。
- 修复：不要解除对dugite的引用，以减少macos的构建大小
- 修复：空json配置文件粉碎应用程序
- 修复：关于没有adminToken的错误
- 修复：导入/导出HTML文档时的错误： 导入/导出HTML wiki时出错： path "参数必须是字符串类型。接收到未定义的（b...
- 修复：Git Worker未捕获的错误粉碎了应用程序，以及未翻译的错误信息
- 修复：休眠的工作区仍有活动视图
- 修正：输入框在wiki显示对话框后失去焦点（调用window.alert('...')）。
- 修复：只在创建新维基时设置语言（克隆时不设置）。
- 修正：防止在某些罕见情况下webContents为空
- 修复：保存和恢复菜单栏窗口位置
- 修复：当应用程序在后台时，sendToMainWindowAndAwait错误。
- 修复：某些配置没有触发重启的问题
- 修复：删除维基后，视图仍然有效
- 修复：缩放菜单项（ctrl+ +/- 缩放）。
- 修复：zx脚本不工作
- 重构：从`src/preload/wikiOperation.ts`为wikiOperations添加更多操作
- 重构：可以使用文件协议为链接打开文件夹并加载图片，由 @linonetwo 在 https://github.com/tiddly-gittly/TidGi-Desktop/pull/421
- 重构：使adminToken的使用更加清晰，以修复潜在的错误
- 重构：将博客相关的选项移到工作区设置的手风琴上
- 重构：在wiki方法中使用workspaceID而不是wikiFolderLocation（这可能导致bug，需要更多测试）。

**Full Changelog**: https://github.com/tiddly-gittly/TidGi-Desktop/compare/v0.7.14...v0.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/tiddly-gittly/TidGi-Desktop/releases/tag/v0.8.0'>v0.8.0</a>.</em>

**Comment by wwhiffle** [comment]
v0.8.0版本的tidgi在Windows x64上重启服务和刷新的时间明显变长

- 测试的重启服务和刷新时间
  - 安装时附带的wiki模板——3.393秒
  - 个人wiki——5.142秒

以往的版本基本在两秒左右重启服务完成，希望在后续版本可以继续优化。

**Reply by linonetwo** [reply]
我也发现了，看性能监测主要是花在了 HTML Parse 上，很奇怪。这次是用了 Electron 新出的 https://www.electronjs.org/docs/latest/api/protocol#protocolhandlescheme-handler 功能，不知为啥会比纯 HTML 更慢，理论上其实是应该更快很多才对。
